### PR TITLE
Add missing json reader options for JsonScanRetrySuite

### PR DESCRIPTION
**Pull request by jihoonson**
Fixes https://github.com/NVIDIA/spark-rapids/issues/11897. `JsonScanRetrySuite` currently creates a JSON reader options that is not compatible to what the plugin uses. This PR fixes it by consolidating the logic to create a base builder for the options.

### CODE REVIEW

File: tests/src/test/scala/com/nvidia/spark/rapids/JsonScanRetrySuite.scala [hunk 1]
```diff
@@ -29,7 +29,7 @@ class JsonScanRetrySuite extends RmmSparkRetrySuiteBase {
 
     val cudfSchema = GpuColumnVector.from(StructType(Seq(StructField("a", IntegerType),
       StructField("b", IntegerType))))
-    val opts = JSONOptions.builder().withLines(true).build()
+    val opts = GpuJsonReadCommon.baseCudfJsonOptionsBuilder().withLines(true).build()
```
Comment (ttnghia) [inline tests/src/test/scala/com/nvidia/spark/rapids/JsonScanRetrySuite.scala:32]: `line(true` is default and the plugin relies on it. So probably we should better set `withLines(true)` in `baseCudfJsonOptionsBuilder` instead.

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonReadCommon.scala [hunk 2]
```diff
@@ -193,18 +193,22 @@ object GpuJsonReadCommon {
     val allowUnquotedControlChars = options.buildJsonFactory()
       .isEnabled(JsonParser.Feature.ALLOW_UNQUOTED_CONTROL_CHARS)
 
+    baseCudfJsonOptionsBuilder()
```
Comment (ttnghia) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonReadCommon.scala:196]: I feel that this name can be something better but don't know what :smiley:

File: tests/src/test/scala/com/nvidia/spark/rapids/JsonScanRetrySuite.scala (hunk 1 above)
Comment (jihoonson) [reply tests/src/test/scala/com/nvidia/spark/rapids/JsonScanRetrySuite.scala:32]: I think we should do that in a separate PR. I'd like to keep the change of this PR minimized.

Comment (ttnghia) [reply tests/src/test/scala/com/nvidia/spark/rapids/JsonScanRetrySuite.scala:32]: But you already changed the `cudfJsonOptions` function right? So adding one more change like that would not be different IMO.

Comment (jihoonson) [reply tests/src/test/scala/com/nvidia/spark/rapids/JsonScanRetrySuite.scala:32]: I am modifying the code in this function without functionality change. Your suggestion changes its behavior which may or may not have an impact on other tests. Given that this test failure is blocking all PRs, I think it's better to make that change in a different PR without unnecessarily blocking others further.

**Comment by jihoonson** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonReadCommon.scala (hunk 2 above)
Comment (jihoonson) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonReadCommon.scala:196]: Can you elaborate? From what perspective can it be better? Like readability? brevity?

Comment (ttnghia) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonReadCommon.scala:196]: What is "base" and why? The builder is not final and not a "base" for something. It is just the builder, with some options set, and before we pass in more options. So I don't like to have "base" in it. But I don't have a better name for it.

Comment (jihoonson) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonReadCommon.scala:196]: I see. I used "base" to imply that it has default options. How about `cudfJsonOptionsBuilderWithDefaults()`? Is that better? It's little long though.

Comment (ttnghia) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonReadCommon.scala:196]: `defaultJsonOptionsBuilder`? No need to have such long name.

File: tests/src/test/scala/com/nvidia/spark/rapids/JsonScanRetrySuite.scala (hunk 1 above)
Comment (jihoonson) [reply tests/src/test/scala/com/nvidia/spark/rapids/JsonScanRetrySuite.scala:32]: Created https://github.com/NVIDIA/spark-rapids/issues/11899.

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonReadCommon.scala (hunk 2 above)
Comment (jihoonson) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonReadCommon.scala:196]: Thanks. I like the name, but want to have `cudf` in it as there are two different `JSONOptions`, one for cudf and another for catalyst. I will rename it to `defaultCudfJsonOptionsBuilder` in https://github.com/NVIDIA/spark-rapids/issues/11899.

**Comment by abellina** [comment]
The main context that I am missing is why we need to disable features in the json reader at all (and all the features were disabled are allowing something to happen, but it seems they allow exceptions???). I was expecting to see disallowing "throwing exceptions".  If we do need this special case for the test, we should put document it in the test/builder to say why it is configured this way.

I'll +1 this for now to unblock CI. I think we just need to have a follow on to clean up docs.

**Review by abellina — APPROVED** [review]
@jihoonson @ttnghia thanks for looking at this

**Review by pxLi — APPROVED** [review]
+1 to unblock CI thanks

**Pull request merged**